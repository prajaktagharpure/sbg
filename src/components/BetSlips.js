import React, { useEffect } from 'react'
import PropTypes from 'prop-types'
import U from '../utils/Util'

const BetSlips = function BetSlips({ betSlips, handleRemove }) {
  useEffect(() => {}, [])

  return (
    <ul className='collection with-header'>
      <li className='collection-header'>
        <h4>My BetSlip</h4>
      </li>
      {!U.isArrayEmpty(betSlips) &&
        betSlips.map((betSlip, key) => (
          <li className='collection-item' key={'bs-' + key}>
            {betSlip.name}
            <i
              id={'bs-' + key}
              className='material-icons right'
              onClick={handleRemove.bind(this)}>
              clear
            </i>
          </li>
        ))}
      {U.isArrayEmpty(betSlips) && (
        <li className='collection-item' key='bs-none'>
          No BetSlips Added
        </li>
      )}
    </ul>
  )
}

BetSlips.propTypes = {
  betSlips: PropTypes.array.isRequired,
  handleRemove: PropTypes.func.isRequired
}

export default BetSlips
