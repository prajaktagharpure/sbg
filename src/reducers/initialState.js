export default {
  events: [],
  market: [],
  eventFullDetails: [],
  betSlips: []
}
