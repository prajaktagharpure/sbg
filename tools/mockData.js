const sportEvents = [
  {
    eventId: 21249939,
    name: 'Shanghai Shenhua 0 v 0 Shandong Luneng Taishan',
    displayOrder: -1000,
    sort: 'MTCH',
    linkedEventId: 21228740,
    classId: 5,
    className: 'Football',
    typeId: 10003971,
    typeName: 'Football Live',
    linkedEventTypeId: 10005942,
    linkedEventTypeName: 'Chinese Super League',
    startTime: '2017-09-19T11:35:23.000Z',
    scores: { home: 0, away: 0 },
    competitors: [
      { name: 'Shanghai Shenhua', position: 'home' },
      { name: 'Shandong Luneng Taishan', position: 'away' }
    ],
    status: {
      active: true,
      started: true,
      live: true,
      resulted: false,
      finished: false,
      cashoutable: true,
      displayable: true,
      suspended: false,
      requestabet: false
    },
    boostCount: 0,
    superBoostCount: 0
  },
  {
    eventId: 21249937,
    name: 'Home United 0 v 1 Albirex Niigata FC',
    displayOrder: 100,
    sort: 'MTCH',
    linkedEventId: 21249693,
    classId: 5,
    className: 'Football',
    typeId: 10003971,
    typeName: 'Football Live',
    linkedEventTypeId: 10005913,
    linkedEventTypeName: 'Singapore S League',
    startTime: '2017-09-19T11:29:23.000Z',
    scores: { home: 0, away: 1 },
    competitors: [
      { name: 'Home United', position: 'home' },
      { name: 'Albirex Niigata FC', position: 'away' }
    ],
    status: {
      active: true,
      started: true,
      live: true,
      resulted: false,
      finished: false,
      cashoutable: true,
      displayable: true,
      suspended: false,
      requestabet: false
    },
    boostCount: 0,
    superBoostCount: 0
  }
]

const eventDetails = {
  eventId: 21249939,
  name: 'Shanghai Shenhua 0 v 0 Shandong Luneng Taishan',
  displayOrder: -1000,
  sort: 'MTCH',
  linkedEventId: 21228740,
  classId: 5,
  className: 'Football',
  typeId: 10003971,
  typeName: 'Football Live',
  linkedEventTypeId: 10005942,
  linkedEventTypeName: 'Chinese Super League',
  startTime: '2017-09-19T11:35:23.000Z',
  scores: { home: 0, away: 0 },
  competitors: [
    { name: 'Shanghai Shenhua', position: 'home' },
    { name: 'Shandong Luneng Taishan', position: 'away' }
  ],
  status: {
    active: true,
    started: true,
    live: true,
    resulted: false,
    finished: false,
    cashoutable: true,
    displayable: true,
    suspended: false,
    requestabet: false
  },
  boostCount: 0,
  superBoostCount: 0
}

const eventFullDetails = [
  {
    event: {
      eventId: 21249939,
      name: 'Shanghai Shenhua 0 v 0 Shandong Luneng Taishan',
      displayOrder: -1000,
      sort: 'MTCH',
      linkedEventId: 21228740,
      classId: 5,
      className: 'Football',
      typeId: 10003971,
      typeName: 'Football Live',
      linkedEventTypeId: 10005942,
      linkedEventTypeName: 'Chinese Super League',
      startTime: '2017-09-19T11:35:23.000Z',
      scores: { home: 0, away: 0 },
      competitors: [
        { name: 'Shanghai Shenhua', position: 'home' },
        { name: 'Shandong Luneng Taishan', position: 'away' }
      ],
      status: {
        active: true,
        started: true,
        live: true,
        resulted: false,
        finished: false,
        cashoutable: true,
        displayable: true,
        suspended: false,
        requestabet: false
      },
      boostCount: 0,
      superBoostCount: 0,
      markets: [
        93775335,
        93775336,
        93775337,
        93775338,
        93775339,
        93777094,
        93777375,
        93777385,
        93777503,
        93777682,
        93649785,
        93649830,
        93767581,
        93775329,
        93775330
      ]
    }
  },
  {
    event: {
      eventId: 21249939,
      name: 'Shanghai Shenhua 0 v 0 Shandong Luneng Taishan',
      displayOrder: -1000,
      sort: 'MTCH',
      linkedEventId: 21228740,
      classId: 5,
      className: 'Football',
      typeId: 10003971,
      typeName: 'Football Live',
      linkedEventTypeId: 10005942,
      linkedEventTypeName: 'Chinese Super League',
      startTime: '2017-09-19T11:35:23.000Z',
      scores: { home: 0, away: 0 },
      competitors: [
        { name: 'Shanghai Shenhua', position: 'home' },
        { name: 'Shandong Luneng Taishan', position: 'away' }
      ],
      status: {
        active: true,
        started: true,
        live: true,
        resulted: false,
        finished: false,
        cashoutable: true,
        displayable: true,
        suspended: false,
        requestabet: false
      },
      boostCount: 0,
      superBoostCount: 0,
      markets: [
        93775335,
        93775336,
        93775337,
        93775338,
        93775339,
        93777094,
        93777375,
        93777385,
        93777503,
        93777682,
        93649785,
        93649830,
        93767581,
        93775329,
        93775330
      ]
    }
  }
]

const betSlips = [
  {
    eventName: 'Kuching FA 0 v 1 UKM FC',
    marketName: 'Half-Time/Full-Time',
    name: 'Half-Time/Full-Time - Kuching FA/Kuching FA',
    outcome: {
      displayOrder: 10,
      eventId: 21249944,
      marketId: 93649434,
      name: 'Kuching FA/Kuching FA',
      outcomeId: 367529109,
      price: { decimal: '5.5', num: '9', den: '2' },
      result: { place: 0, result: '-', favourite: false },
      status: {
        active: true,
        resulted: false,
        cashoutable: false,
        displayable: true,
        suspended: false
      }
    }
  }
]

const currEventData = eventFullDetails.slice()[0]

// Using CommonJS style export so we can consume via Node (without using Babel-node)
module.exports = {
  sportEvents,
  eventFullDetails,
  eventDetails,
  betSlips
}
