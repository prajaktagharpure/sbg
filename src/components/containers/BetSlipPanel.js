import React from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'
import BetSlips from '../BetSlips'
import { removeBetSlip } from '../../actions/betSlipsActions'
import { bindActionCreators } from 'redux'

export function BetSlipPanel({ betSlips, actions }) {
  function handleRemove(e) {
    const index = parseInt(e.target.id.split('-')[1])
    actions.removeBetSlip(index)
  }

  return <BetSlips betSlips={betSlips} handleRemove={handleRemove} />
}

BetSlipPanel.propTypes = {
  betSlips: PropTypes.array.isRequired,
  actions: PropTypes.object.isRequired
}

function mapStateToProps(state) {
  return {
    betSlips: [...state.betSlips]
  }
}

function mapDispatchToProps(dispatch) {
  return {
    actions: {
      removeBetSlip: bindActionCreators(removeBetSlip, dispatch)
    }
  }
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(BetSlipPanel)
