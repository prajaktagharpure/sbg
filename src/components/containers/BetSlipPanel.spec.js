import React from 'react'
import { betSlips } from '../../../tools/mockData'
import { BetSlipPanel } from './BetSlipPanel'
import BetSlips from '../BetSlips'
import { shallow } from 'enzyme'
import configureMockStore from 'redux-mock-store'
import { Provider } from 'react-redux'
import { create } from 'react-test-renderer'
import initialState from '../../reducers/initialState'

describe('<BetSlipPanel />', () => {
  const props = {
    betSlips,
    actions: {
      removeBetSlip: jest.fn()
    }
  }
  const wrapper = shallow(<BetSlipPanel {...props} />)

  it('should contain <BetSlips />', () => {
    expect(wrapper.find(BetSlips).length).toEqual(1)
  })

  it('should match snapshot', () => {
    const props = {
      betSlips,
      actions: {
        removeBetSlip: jest.fn()
      }
    }
    const store = configureMockStore()(initialState)
    const component = create(
      <Provider store={store}>
        <BetSlipPanel {...props} />
      </Provider>
    )
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })
})
