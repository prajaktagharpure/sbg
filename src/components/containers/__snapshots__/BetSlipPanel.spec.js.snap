// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BetSlipPanel /> should match snapshot 1`] = `
<ul
  className="collection with-header"
>
  <li
    className="collection-header"
  >
    <h4>
      My BetSlip
    </h4>
  </li>
  <li
    className="collection-item"
  >
    Half-Time/Full-Time - Kuching FA/Kuching FA
    <i
      className="material-icons right"
      id="bs-0"
      onClick={[Function]}
    >
      clear
    </i>
  </li>
</ul>
`;
